source/test/TestDirector.cc

```diff
@@ -0,0 +1,101 @@
+#include "TestDirector.hh"
+
+#include <game/kart/KartObjectManager.hh>
+
+#include <abstract/File.hh>
+
+#include <cstddef>
+
+namespace Test {
+
+TestDirector::TestDirector() {
+    size_t size;
+    u8 *file = Abstract::File::Load("Test/rmc3-rta-1-17-843.krkg", size);
+    m_file = file;
+    m_stream = EGG::RamStream(file, static_cast<u32>(size));
+    m_currentFrame = -1;
+    m_sync = true;
+
+    // Initialize endianness for the RAM stream
+    u16 mark = *reinterpret_cast<u16 *>(file + offsetof(TestHeader, byteOrderMark));
+    std::endian endian = parse<u16>(mark) == 0xfeff ? std::endian::big : std::endian::little;
+    m_stream.setEndian(endian);
+
+    readHeader();
+}
+
+TestDirector::~TestDirector() = default;
+
+bool TestDirector::init() {
+    assert(m_stream.read_u32() == m_stream.index());
+    return calc();
```

Comment:
This means that `calc` will compute a frame before we enter the frame loop in `KSystem::run`. Is this intended?

Reply:
Disregard